Pod/Illuminator/IlluminatorTestProgress.swift

```diff
@@ -9,6 +9,42 @@
 import Foundation
 import XCTest
 
+/*
+ * This protocol allows custom handling of results.  For example, screenshots of failure states or desktop notifications
+ * isPass and isFail can both be false -- it indicates a "Flagging" state.  They are guaranteed to not both be true
+ */
+public protocol IlluminatorTestResultHandler {
+    associatedtype AbstractStateType
```

Comment:
We dont need to bind to associatedtype AbstractStateType because there is no requirement for any AbstractStateType functionality...  method should be enough